### DISCUSSION TITLE
xrandr output names keep switching

### DISCUSSION THREAD

**Discussion by erahhal** [i3/i3#4830] [Q&A]
I've got two identical displays connected to a thunderbolt hub with two DP ports.  When plugging them in, they frequently switch xrandr output names, one being DP-1, the other as DP-2, and vice versa.  This causes them to switch visual order all the time.  I can use `autorandr` to get the displays in the right order based on EDID, but is there a way to order workspaces in i3 using the EDID or xrandr position rather than xrandr output name, which is not stable?

**Comment by bugi** [comment]
In my case, I was able to tell my monitor to use DP 1.0a instead of 1.2 and all was well.  (And plan to try something similar to [this autorandr+i3-msg scheme](https://www.reddit.com/r/i3wm/comments/hjiwtv/is_there_a_way_to_have_different_workspace_output/).)

Is this a good place for a feature request?  Make i3 _output_ clauses accept an EDID.  That plus autorandr would probably be a good solution for me.

The displayport specs require that unstable behavior.  I ran into it using i3+AMD and found [this discussion around nvidia hardware with mswindows](https://www.reddit.com/r/nvidia/comments/66opvp/this_displayport_hotplug_madness_finally_needs_to/) enlightening.  The short version is that it's a known issue; everybody hates it; there's both hardware and software workarounds.
